tests/api/ping.rs

```diff
@@ -0,0 +1,195 @@
+use crate::helpers::app::test_app_with_login;
+use crate::helpers::db::reset;
+use crate::helpers::http_client::{PostPayload, TestHttpClient};
+use crate::helpers::wait_for_stubr;
+use actix_web::test;
+use anyhow::Error;
+use diesel::prelude::*;
+use rumba::db::schema;
+use serde_json::{json, Value};
+
+#[actix_rt::test]
+#[stubr::mock(port = 4321)]
+async fn test_empty() -> Result<(), Error> {
+    let pool = reset()?;
+    wait_for_stubr().await?;
+
+    let app = test_app_with_login(&pool).await?;
+    let service = test::init_service(app).await;
+    let mut logged_in_client = TestHttpClient::new(service).await;
+
+    let res = logged_in_client
+        .post("/api/v1/ping", None, Some(PostPayload::Form(json!({}))))
+        .await;
+    assert_eq!(res.response().status(), 200);
+
+    let mut conn = pool.get()?;
+    let activity_data = schema::activity_pings::table
+        .select(schema::activity_pings::activity)
+        .first::<Value>(&mut conn)?;
+    assert_eq!(activity_data, json!({ "subscription_type": "mdn_plus_5m" }));
+
+    drop(stubr);
+    Ok(())
+}
+
+#[actix_rt::test]
+#[stubr::mock(port = 4321)]
+async fn test_garbage() -> Result<(), Error> {
+    let pool = reset()?;
+    wait_for_stubr().await?;
+
+    let app = test_app_with_login(&pool).await?;
+    let service = test::init_service(app).await;
+    let mut logged_in_client = TestHttpClient::new(service).await;
+
+    let res = logged_in_client
+        .post(
+            "/api/v1/ping",
+            None,
+            Some(PostPayload::Form(json!({ "foobar": "barfoo" }))),
+        )
+        .await;
+    assert_eq!(res.response().status(), 200);
```

Comment:
Is it really okay if there is garbage in there? We are creating a needless row of database for the sake of nothing. Should we be a little stricter with valid fields? I think it would help catch any breaking changes later.

Reply:
To be clear: in this test we POST garbage, but ensure that none of it gets saved in the db - the activity column is an empty json object. The row is still created (or updated) as it's still a valid ping indicating usage from the client side.

I foresee "garbage" being potential future pings that we've added to yari which haven't made their way to rumba yet - this test ensures things "fail" gracefully.